libs/elasticsearch-core/build.gradle

```diff
@@ -52,6 +52,29 @@ forbiddenApisMain {
     signaturesURLs = [PrecommitTasks.getResource('/forbidden/jdk-signatures.txt')]
 }
 
+// we want to keep the JDKs in our IDEs set to JDK 8 until minimum JDK is bumped to 9 so we do not include this source set in our IDEs
+if (!isEclipse && !isIdea) {
+    sourceSets {
+        java9 {
```

Comment:
Technically this is java10 now, I think.

Reply:
Wouldn't that mean someone running with JDK 9 wouldn't use the 9-specific one? The code only uses Java 9 features so far so I figured that's where we should stick for now

Reply:
You can't build Elasticsearch with Java 9 as of last week I believe. That is all I'm getting at.  I dunno if we want to bump all the java9 directories to java10 because of that. I think *probably*, but for now we can keep it.

Reply:
Even with java 10, we are still building this code with java 9 *source* level. It's really only about the standard features/libs that are allowed when compiling.